pkg/windowsperfcountercommon/scraper.go

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+// +build windows
+
+package windowsperfcountercommon // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/windowsperfcountercommon"
+
+import (
+	"fmt"
+
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/windowsperfcountercommon/internal/pdh"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/windowsperfcountercommon/internal/third_party/telegraf/win_perf_counters"
+)
+
+//
+type PerfCounterScraper interface {
+	// Path returns the counter path
+	Path() string
+	// ScrapeData collects a measurement and returns the value(s).
+	ScrapeData() ([]win_perf_counters.CounterValue, error)
+	// Close all counters/handles related to the query and free all associated memory.
+	Close() error
+}
+
+const instanceLabelName = "instance"
+
+type ScraperCfg struct {
+	CounterCfg PerfCounterConfig
+	Metric     MetricRep
+}
+
+type MetricRep struct {
+	Name       string
+	Attributes map[string]string
+}
+
+type Scraper struct {
```

Comment:
Using the term Scraper in this package is a little confusing to me, since we'll have scrapers managing scrapers. Any other ideas? The best I can think of offhand is "Watcher".

Reply:
Updated